clinica/iotools/converters/adni_to_bids/adni_utils.py

```diff
@@ -76,46 +101,81 @@ def visits_to_timepoints(
-            preferred_visit_name = f"ADNI3 Year {year} Visit"
-    elif visit.ORIGPROT == "ADNI2":
-        if visit.VISCODE == "bl":
-            preferred_visit_name = "ADNI2 Screening MRI-New Pt"
-        elif visit.VISCODE == "m03":
-            preferred_visit_name = "ADNI2 Month 3 MRI-New Pt"
-        elif visit.VISCODE == "m06":
-            preferred_visit_name = "ADNI2 Month 6-New Pt"
-        else:
-            year = str(int(visit.VISCODE[1:]) / 12)
-            preferred_visit_name = f"ADNI2 Year {year} Visit"
-    else:
-        if visit.VISCODE == "bl":
-            if visit.ORIGPROT == "ADNI1":
-                preferred_visit_name = "ADNI Screening"
-            else:  # ADNIGO
-                preferred_visit_name = "ADNIGO Screening MRI"
-        elif visit.VISCODE == "m03":  # Only for ADNIGO Month 3
-            preferred_visit_name = "ADNIGO Month 3 MRI"
-        else:
-            month = int(visit.VISCODE[1:])
-            if month < 54:
-                preferred_visit_name = f"ADNI1/GO Month {str(month)}"
-            else:
-                preferred_visit_name = f"ADNIGO Month {str(month)}"
-    return preferred_visit_name
+    if study == ADNIStudy.ADNI3:
+        return _get_preferred_visit_name_adni3(visit_code)
+    if study == ADNIStudy.ADNI2:
+        return _get_preferred_visit_name_adni2(visit_code)
+    if study == ADNIStudy.ADNI1:
+        return _get_preferred_visit_name_adni1(visit_code)
+    if study == ADNIStudy.ADNIGO:
+        return _get_preferred_visit_name_adnigo(visit_code)
+
+
+def _get_preferred_visit_name_adni3(visit_code: str) -> str:
+    if visit_code == "bl":
+        return "ADNI Screening"
+    return f"ADNI3 Year {_parse_year_from_visit_code(visit_code)} Visit"
+
+
+def _parse_year_from_visit_code(visit_code: str) -> float:
+    """Return the year corresponding to the visit code.
+    Assumes a visit code of the form 'mXXX'.
+    """
+    return _parse_month_from_visit_code(visit_code) / 12
+
+
+def _parse_month_from_visit_code(visit_code: str) -> int:
+    """Return the month corresponding to the visit code.
+    Assumes a visit code of the form 'mXXX'.
+    """
+    try:
+        return int(visit_code[1:])
+    except Exception:
+        raise ValueError(
+            f"Cannot extract month from visit code {visit_code}."
+            "Expected a code of the form 'mXXX' where XXX can be casted to an integer."
+        )
+
+
+def _get_preferred_visit_name_adni2(visit_code: str) -> str:
+    if visit_code == "bl":
+        return "ADNI2 Screening MRI-New Pt"
+    if visit_code == "m03":
+        return "ADNI2 Month 3 MRI-New Pt"
+    if visit_code == "m06":
+        return "ADNI2 Month 6-New Pt"
+    return f"ADNI2 Year {_parse_year_from_visit_code(visit_code)} Visit"
+
+
+def _get_preferred_visit_name_adni1(visit_code: str) -> str:
+    if visit_code == "bl":
+        return "ADNI Screening"
+    if visit_code == "m03":
+        return "ADNIGO Month 3 MRI"  # does not make any sense...
```

Comment:
This is also a bit weird. I think that this relies on the fact that 'm03' is never found for ADNI1 ??